packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx

```diff
@@ -826,31 +817,154 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
   };
 
   handleStateChange = (changes: StateChangeOptions<ShorthandValue<DropdownItemProps>>) => {
-    if (changes.isOpen !== undefined && changes.isOpen !== this.state.open) {
-      const newState = { open: changes.isOpen, highlightedIndex: this.state.highlightedIndex };
-
-      if (changes.isOpen) {
-        const highlightedIndexOnArrowKeyOpen = this.getHighlightedIndexOnArrowKeyOpen(changes);
-        if (_.isNumber(highlightedIndexOnArrowKeyOpen)) {
-          newState.highlightedIndex = highlightedIndexOnArrowKeyOpen;
+    const { search, multiple, highlightFirstItemOnOpen, items, getA11ySelectionMessage } = this.props;
+    const { value, open } = this.state;
+    const {
+      type,
+      inputValue: newSearchQuery,
+      selectedItem: newValue,
+      isOpen: newOpen,
+      highlightedIndex: newHighlightedIndex,
+    } = changes;
```

Comment:
nit: I don't think that we should destructure there as it hard to follow that comes from `props`/`state` & `downshift.changes`

Reply:
reverted this and used changes.variable. thanks!